src/main/java/io/r2dbc/postgresql/util/Assert.java

```diff
@@ -133,4 +134,28 @@ public static String requireFileExistsOrNull(@Nullable String file, String messa
         return file;
     }
 
+    /**
+     * Checks that the specified value is of a specific type.
+     *
+     * @param value     the value to check
+     * @param predicate predicate to check value
+     * @param message   the message to use in exception if type is not as required
+     * @param <T>       the type being required
+     * @return the value casted to the required type
+     * @throws IllegalArgumentException if {@code value} is not of the required type
+     * @throws IllegalArgumentException if {@code value}, {@code type}, or {@code message} is {@code null}
+     */
+    public static <T> T require(T value, Predicate<T> predicate, String message) {
```

Comment:
Typically, we have something like [`Assert.isTrue(…)`](https://github.com/spring-projects/spring-framework/blob/master/spring-core/src/main/java/org/springframework/util/Assert.java#L116). Spring 5 introduced also variants that accept a `Supplier<String> messageSupplier` so that messages using `String.format(…)` don't get executed each time the method gets hit.